bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java

```diff
@@ -355,15 +355,26 @@ private static String getUserPersistenceDataFolder() {
     public void activate(final Map<String, Object> config) {
         // add default configurations
         RrdDefConfig defaultNumeric = new RrdDefConfig("default_numeric");
-        defaultNumeric.setDef("GAUGE,60,U,U,60");
-        defaultNumeric.addArchives(
-                "AVERAGE,0.5,1,480:AVERAGE,0.5,4,360:AVERAGE,0.5,14,644:AVERAGE,0.5,60,720:AVERAGE,0.5,720,730:AVERAGE,0.5,10080,520");
+        // use 10 seconds as a step size for numeric values and allow a 10 minute silence between updates
+        defaultNumeric.setDef("GAUGE,600,U,U,10");
+        // define 5 different boxes:
+        // 1. granularity of 10s for the last hour
+        // 2. granularity of 1m for the last week
+        // 3. granularity of 15m for the last year
+        // 4. granularity of 1h for the last 5 years
+        // 5. granularity of 1d for the last 10 years
+        defaultNumeric.addArchives("AVERAGE,0.5,1,360:AVERAGE,0.5,6,10080:AVERAGE,0.5,90,36500:AVERAGE,0.5,8640,3650");
```

Comment:
I would like to suggest FIRST or LAST for numerical values too.
The reason is, I'm contemplating aggregates series in the UI by hour/day, to make it possible to build calendar or weekly displays like these:
- https://echarts.apache.org/examples/en/editor.html?c=calendar-heatmap
- https://echarts.apache.org/examples/en/editor.html?c=heatmap-cartesian
- https://echarts.apache.org/examples/en/editor.html?c=scatter-single-axis
- https://echarts.apache.org/examples/en/editor.html?c=scatter-punchCard
- https://echarts.apache.org/examples/en/editor.html?c=scatter-polar-punchCard
 
One of these aggregations would be the difference to the previous period: it would itself collect points found in that hour or day slice, take the first or last one and subtract the result of the previous period to it. That way if you have for instance an energy consumption metric you could get the hourly/daily consumption more accurately than by processing averaged data.
Does that make sense?

Reply:
> One of these aggregations would be the difference to the previous period

Which is exactly, what rrd4j is doing with the [COUNTER datasource type](https://www.openhab.org/addons/persistence/rrd4j/#datasource-types), correct?
I actually never used the COUNTER myself and it makes sense to me to rather persist the absolute number and do the diffs only when creating the charts (as it can then be flexible wrt interval).

>  you could get the hourly/daily consumption more accurately than by processing averaged data.

Yes, it would be more accurate, but I think one could also live with the averages.
If you think of sensor values, using FIRST or LAST will in contrast have a negative effect on the accuracy: The whole graph will either be slightly too left or too right, while the average would make sure that the lines represent the value at a given point in time. Also, outliers can have a stronger negative effect, in case they are the first or last value of a given interval.
I am not sure how bad those effects are and they probably depend on the granularity. I'd be ok to give it a try, if you wish.

A different strong reason that I see to not use AVERAGE for Number items are items where the states are not sensor values, but merely discrete enumerations - using average will create illegal states for those.

Reply:
> The whole graph will either be slightly too left or too right, while the average would make sure that the lines represent the value at a given point in time. Also, outliers can have a stronger negative effect, in case they are the first or last value of a given interval.

Fair point.

> I am not sure how bad those effects are and they probably depend on the granularity.

With the kind of granularity you're proposing here, you would still have 24 data points per day for the last 5 years, so if we're building an temperature calendar for example, averaging these 24 points would imho give a value rather close to the actual average of the day. If you're aggregating by hour, you'd still have 4 data points for the last year.

Beyond these periods, it's becoming problematic indeed, like the only value retained for a day would be one close to midnight which is obviously not a good value to get an average, but it's 5 years in the past... I think that's a trade-off we have to take to get good enough results while avoiding users to get into their persistence strategies - which remains an option anyway.

Reply:
A little bit late to the party, but LAST is IMO a bad choice for sensor values. AVERAGE is not only the better choice for values in the past but also for values which are a bit flaky because of the measurement involved.

Reply:
@J-N-K Did you read through all our discussion? If I understand you correct, your point is the same as what I mentioned about the outliers. I see some risk there, but would nonetheless give it a try.

Reply:
Would it be possible to use the item's dimension? Like a dimensionless number would be LAST, as would e.g. Energy, while other quantity measurements would be averaged.

Reply:
I also thought about the same, but didn't yet think it through. We could indeed create different defs for "plain" Number items and for Number items with a dimension. Anything with a dimension should hopefully be measurable and an average thus makes sense. The "plain" items could be anything and LAST is thus the safer option.

Reply:
An energy meter is a perfect example for a COUNTER. If you get 34567.122 kWh, multiply by thousand and store in a COUNTER. 

@kaikreuzer yes. But I was not only thinking of outliers but of values that have a large statistical variance. Imagine someone (not me but I‘ve seen people use that) who meters radioactivity. He will for sure not store the total number of counts (That‘s a useless value, If total dose matters we have other problems than data storage.), he wants to store „per time“. So his unit would be Bequerel and measure every minute. Each of these values will be very different (because count rate is low in any case). The average is a much better aggregation than last value, and that is true after an hour.

This is also true for fast changing temperatures (like central heating boiler temperature). Aggregating over an 15 minutes might lose turn-ons (turns on immediately at the beginning, runs for 5 minutes, temperature is nearly back to normal after 15 minutes, when the next LAST value is taken).